core/templates/dev/head/pages/topic_viewer/stories_list_directive.html

```diff
@@ -0,0 +1,98 @@
+<div ng-if="canonicalStories.length === 0">
+  <em>There are no stories to display.</em>
+</div>
+
+<div ng-if="!topicViewerWindowIsNarrow">
+  <md-button class="md-no-ink oppia-topic-viewer-carousel-scroller"
+             ng-click="scroll(true)"
+             ng-hide="leftmostCardIndices == 0">
+    <i class="material-icons">&#xE314;</i>
+  </md-button>
+  <div class="oppia-topic-viewer-carousel-scroller" ng-if="leftmostCardIndices == 0" aria-hidden="true">
+  </div>
+
+  <div class="oppia-topic-viewer-carousel"
+       ng-swipe-right="scroll(true)"
+       ng-swipe-left="scroll(false)">
+    <div class="oppia-topic-viewer-carousel-tiles">
+      <md-card ng-repeat="canonicalStory in canonicalStories" class="oppia-story-summary-tile">
+        <story-summary-tile story-id="canonicalStory.id"
+                            title="canonicalStory.title"
+                            description="canonicalStory.description">
+        </story-summary-tile>
+      </md-card>
+    </div>
+  </div>
+
+  <md-button class="md-no-ink oppia-topic-viewer-carousel-scroller"
+             ng-click="scroll(false)"
+             ng-hide="(canonicalStories.length - tileDisplayCount) <= leftmostCardIndices"
+             aria-hidden="true">
+    <i class="material-icons">&#xE315;</i>
+  </md-button>
+  <div class="oppia-topic-viewer-carousel-scroller"
+       ng-if="(canonicalStories.length - tileDisplayCount) <= leftmostCardIndices">
+  </div>
+</div>
+
+<div ng-if="topicViewerWindowIsNarrow" class="oppia-topic-viewer-low-width">
+  <md-card ng-repeat="canonicalStory in canonicalStories" class="oppia-story-summary-tile">
+    <canonical-story-summary-tile story-id="canonicalStory.id"
+                                  title="canonicalStory.title"
+                                  description="canonicalStory.description">
+    </canonical-story-summary-tile>
+  </md-card>
+</div>
+
+<style>
+  .oppia-topic-viewer-low-width {
```

Comment:
Prefix all this with stories-list

@apb7 could we please file this as a lint check todo?

Reply:
Done.

Reply:
Hi @seanlip, we already have an issue for this: #5097, the second point. Thanks!